File: reference/reference.net/db4o-reference-chapters/src/Db4objects.Db4odoc/Refactoring/RefactoringExample.cs
/* Copyright (C) 2004 - 2007 db4objects Inc. http://www.db4o.com */
using System;
using System.IO;
using Db4objects.Db4o;
using Db4objects.Db4o.Config;
using Db4objects.Db4o.Query;

namespace Db4objects.Db4odoc.Refactoring
{
	public class RefactoringExample
	{
		private const string Db4oFileName = "reference.db4o";
	
		public static void Main(string[] args) 
		{
			System.Console.WriteLine("Correct sequence of actions: ");
			SetObjects();
			CheckDB();	
			ChangeClass();
			SetNewObjects();
			RetrievePilotNew();

			/*System.Console.WriteLine("Incorrect sequence of actions: ");
			SetObjects();
			CheckDB();	
			SetNewObjects();
			ChangeClass();
			RetrievePilotNew();*/
		}
		// end Main

		private static void SetObjects()
		{
			File.Delete(Db4oFileName);
			IObjectContainer container = Db4oFactory.OpenFile(Db4oFileName);
			try 
			{
				Pilot pilot = new Pilot("Rubens Barrichello");
				container.Set(pilot);
				pilot = new Pilot("Michael Schumacher");
				container.Set(pilot);
			} 
			finally 
			{
				container.Close();
			}
		}
		// end SetObjects

        private static void CheckDB()
		{
			IObjectContainer container = Db4oFactory.OpenFile(Db4oFileName);
			try 
			{
				IObjectSet result = container.Get(typeof(Pilot));
				foreach (object obj in result)
				{
					Pilot pilot = (Pilot)obj;
					System.Console.WriteLine("Pilot="+ pilot);
				}
			} 
			finally 
			{
				container.Close();
			}
		}
		// end CheckDB

        private static void SetNewObjects()
		{
			IObjectContainer container = Db4oFactory.OpenFile(Db4oFileName);
			try 
			{
				PilotNew pilot = new PilotNew("Rubens Barrichello",99);
				container.Set(pilot);
				pilot = new PilotNew("Michael Schumacher",100);
				container.Set(pilot);
			} 
			finally 
			{
				container.Close();
			}
		}
		// end SetNewObjects

        private static void ChangeClass()
		{
            IConfiguration configuration = Db4oFactory.NewConfiguration();
            configuration.ObjectClass(typeof(Pilot)).Rename("Db4objects.Db4odoc.Refactoring.PilotNew, Db4objects.Db4odoc");
            configuration.ObjectClass(typeof(PilotNew)).ObjectField("_name").Rename("_identity");
            IObjectContainer container = Db4oFactory.OpenFile(configuration, Db4oFileName);
            container.Close();
		}
		// end ChangeClass

        private static void RetrievePilotNew()
		{
			IObjectContainer container = Db4oFactory.OpenFile(Db4oFileName);
			try 
			{
				IQuery q = container.Query();
				q.Constrain(typeof(PilotNew));
				IObjectSet result = q.Execute();
				foreach (object obj in result)
				{
					PilotNew pilot = (PilotNew)obj;
					System.Console.WriteLine("Pilot="+ pilot);
				} 
			}
			finally 
			{
				container.Close();
			}
		}
		// end RetrievePilotNew
	}
}

File: reference/reference.vb.net/Db4objects.Db4odoc/Refactoring/RefactoringExample.vb
' Copyright (C) 2004 - 2007 db4objects Inc. http://www.db4o.com 
Imports System
Imports System.IO
Imports Db4objects.Db4o
Imports Db4objects.Db4o.Config
Imports Db4objects.Db4o.Query

Namespace Db4objects.Db4odoc.Refactoring
    Public Class RefactoringExample
        Private Const Db4oFileName As String = "reference.db4o"

        Public Shared Sub Main(ByVal args() As String)
            System.Console.WriteLine("Correct sequence of actions: ")
            SetObjects()
            CheckDB()
            ChangeClass()
            SetNewObjects()
            RetrievePilotNew()

            '           System.Console.WriteLine("Incorrect sequence of actions: ")
            ' 			SetObjects()
            ' 			CheckDB()	
            ' 			SetNewObjects()
            ' 			ChangeClass()
            ' 			RetrievePilotNew()*/
        End Sub
        ' end Main

        Private Shared Sub SetObjects()
            File.Delete(Db4oFileName)
            Dim container As IObjectContainer = Db4oFactory.OpenFile(Db4oFileName)
            Try
                Dim pilot As Pilot = New Pilot("Rubens Barrichello")
                container.Set(pilot)
                pilot = New Pilot("Michael Schumacher")
                container.Set(pilot)
            Finally
                container.Close()
            End Try
        End Sub
        ' end SetObjects

        Private Shared Sub CheckDB()
            Dim container As IObjectContainer = Db4oFactory.OpenFile(Db4oFileName)
            Try
                Dim result As IObjectSet = container.Get(GetType(Pilot))
                Dim obj As Object
                For Each obj In result
                    Dim pilot As Pilot = CType(obj, Pilot)
                    System.Console.WriteLine("Pilot=" + pilot.ToString())
                Next
            Finally
                container.Close()
            End Try
        End Sub
        ' end CheckDB

        Private Shared Sub SetNewObjects()
            Dim container As IObjectContainer = Db4oFactory.OpenFile(Db4oFileName)
            Try
                Dim pilot As PilotNew = New PilotNew("Rubens Barrichello", 99)
                container.Set(pilot)
                pilot = New PilotNew("Michael Schumacher", 100)
                container.Set(pilot)
            Finally
                container.Close()
            End Try
        End Sub
        ' end SetNewObjects

        Private Shared Sub ChangeClass()
            Dim configuration As IConfiguration = Db4oFactory.NewConfiguration()
            configuration.ObjectClass(GetType(Pilot)).Rename("Db4objects.Db4odoc.Refactoring.PilotNew, Db4objects.Db4odoc")
            configuration.ObjectClass(GetType(PilotNew)).ObjectField("_name").Rename("_identity")
            Dim container As IObjectContainer = Db4oFactory.OpenFile(configuration, Db4oFileName)
            container.Close()
        End Sub
        ' end ChangeClass

        Private Shared Sub RetrievePilotNew()
            Dim container As IObjectContainer = Db4oFactory.OpenFile(Db4oFileName)
            Try
                Dim q As IQuery = container.Query()
                q.Constrain(GetType(PilotNew))
                Dim result As IObjectSet = q.Execute()
                Dim obj As Object
                For Each obj In result
                    Dim pilot As PilotNew = CType(obj, PilotNew)
                    System.Console.WriteLine("Pilot=" + pilot.ToString())
                Next
            Finally
                container.Close()
            End Try
        End Sub
        ' end RetrievePilotNew

    End Class
End Namespace

